src/tt.cpp

```diff
@@ -123,7 +123,7 @@ TTEntry* TranspositionTable::probe(const Key key, bool& found) const {
     const uint16_t key16 = uint16_t(key);  // Use the low 16 bits as key inside the cluster
 
     for (int i = 0; i < ClusterSize; ++i)
-        if (tte[i].key16 == key16 || !tte[i].depth8)
+        if (tte[i].key16 == key16)
             return found = bool(tte[i].depth8), &tte[i];
```

Comment:
> can you example now what the bool(tte[i].depth8) does, what's the typical value?

bool(tte[i].depth8) checks for if the entry is found or not
the trick of the depth offsetting with -7 , is it makes depth 0 truthy from qsearch because int(0) is false..
Now since we use a pointer to the entry we can return the entry but not consider it as found, so I assume the older code, was trying to make the code smarter by forcing where to save the entry.

Reply:
Patch makes no sense, imho. As soon as we find an empty slot, we want to return immediately.

Reply:
but my question is, doesn't that already make an entire cluster non-functioning?
to me it seems like the first cluster is never used for saving here. because the loop to follow bellow starts from 1?

Reply:
It works because the first cluser is set as default replacement in line 130. The loop checks only if one of the other clusters in better.

Reply:
i.e. the entries to return from for a no-found entry, is always coming from replacement strategy below which is never from tte[0], this sounds fine, but at the same time, how can the first cluster ever have entries saved onto it, if we never say the first entry has empty slots and the first cluster never having any keys?

Reply:
Oh sorry indeed that works...

Reply:
Yes so what this patch does is it replaces always based on relative age and never on the bases that an entry is empty..
fascinating to me that it passed